db/views/materialized_patient_summaries_v02.sql

```diff
@@ -0,0 +1,138 @@
+       latest_bp_passport.id                                                                                                     AS latest_bp_passport_id,
+       latest_bp_passport.identifier                                                                                             AS latest_bp_passport_identifier,
+       mh.hypertension                                                                                                           AS hypertension,
+       mh.diabetes                                                                                                               AS diabetes,
+       p.id
+FROM patients p
+         LEFT OUTER JOIN addresses ON addresses.id = p.address_id
+         LEFT OUTER JOIN facilities reg_facility ON reg_facility.id = p.registration_facility_id
+         LEFT OUTER JOIN facilities assigned_facility ON assigned_facility.id = p.assigned_facility_id
+         LEFT OUTER JOIN
+     (SELECT DISTINCT ON (patient_id) *
+      FROM medical_histories
+      WHERE deleted_at IS NULL) AS mh ON mh.patient_id = p.id
+         LEFT OUTER JOIN
+     (SELECT DISTINCT ON (patient_id) *
+      FROM patient_phone_numbers
+      WHERE deleted_at IS NULL
+      ORDER BY patient_id,
+               device_created_at DESC) AS latest_phone_number ON latest_phone_number.patient_id = p.id
+         LEFT OUTER JOIN
+     (SELECT DISTINCT ON (patient_id) *
+      FROM blood_pressures
+      WHERE deleted_at IS NULL
+      ORDER BY patient_id,
+               recorded_at DESC) AS latest_blood_pressure ON latest_blood_pressure.patient_id = p.id
+         LEFT OUTER JOIN facilities latest_blood_pressure_facility
+                         ON latest_blood_pressure_facility.id = latest_blood_pressure.facility_id
+         LEFT OUTER JOIN
+     (SELECT DISTINCT ON (patient_id) *
+      FROM blood_sugars
+      WHERE deleted_at IS NULL
+      ORDER BY patient_id,
+               recorded_at DESC) AS latest_blood_sugar ON latest_blood_sugar.patient_id = p.id
+         LEFT OUTER JOIN facilities latest_blood_sugar_facility
+                         ON latest_blood_sugar_facility.id = latest_blood_sugar.facility_id
+         LEFT OUTER JOIN
+     (SELECT DISTINCT ON (patient_id) *
+      FROM patient_business_identifiers
+      WHERE identifier_type = 'simple_bp_passport'
+        AND deleted_at IS NULL
+      ORDER BY patient_id,
+               device_created_at DESC) AS latest_bp_passport ON latest_bp_passport.patient_id = p.id
+         LEFT OUTER JOIN
+     (SELECT DISTINCT ON (patient_id) *
+      FROM appointments
+      WHERE deleted_at IS NULL
+      ORDER BY patient_id, device_created_at DESC) AS next_scheduled_appointment ON next_scheduled_appointment.patient_id = p.id
+                                                                     AND next_scheduled_appointment.status = 'scheduled'
```

Comment:
This is the only change in this file (Change 2).

Reply:
We should NOT filter by scheduled appointments here, right? If the most-recent-device-created appointment has been marked "visited", then the patient is not overdue. We wouldn't be looking at that appointment if we keep this clause in here, right?

Reply:
This clause applies after the `DISTINCT ON` has been applied. If there is no `scheduled` appointment (only a `visited` appointment) the `next_scheduled_appointment` would be NULL and the line list will not show this patients as overdue.

Reply:
Added a spec [acf7d4c](https://github.com/simpledotorg/simple-server/pull/4237/commits/acf7d4cd4a53b56eb3e4e1b8f8e6fa85d8ebc6d9) for this

Reply:
Thanks for the spec. What about the situation we inspected in real prod data last week? What if a patient today has a "scheduled"  appointment for Sept 7th, and a more recently created "visited" appointment for Sept 1st?

Reply:
This patient wouldn't be counted as overdue. 
- `DISTINCT ON` finds the `visited` appointment as the latest appointment
- The appointment doesn't get included in the left join because it's not `scheduled`
- Patient is not counted as overdue

There's a [unit test](https://github.com/simpledotorg/simple-server/pull/4237/files#diff-b7f8fb63d18f5bca6e2e35e61d918bd6fed3ff9f5fd4e9a2b5d8b9b60ab8318eR171) for this scenario.

Reply:
Perfect 👌